doc/contributing/docs/api.rst

```diff
@@ -0,0 +1,170 @@
+Documenting the API
+===================
+
+This document contains general guidelines for describing the Tarantool API,
+as well as examples and templates.
+
+Style
+-----
+
+Please write as simply as possible. Describe functionality using short sentences in the present simple tense.
+A short sentence consists of no more than two clauses.
+Consider using `LanguageTool <https://languagetool.org/>`_ or `Grammarly <https://www.grammarly.com/>`_
+to check your English.
+For more style-related specifics, consult the :doc:`Language and style </contributing/docs/style>` section.
+
+..  _contributing-docs-api_version:
+
+Indicating the version
+----------------------
+
+For every new feature or parameter, please indicate the version it was introduced in.
+
+With Tarantool features and parameters, use one of the following Sphinx directives:
+
+..  code-block:: rst
+
+    Since :tarantool-release:`2.10.0`.
+    This is a link to the release notes on GitHub.
+
+    Since :doc:`2.10.0 </release/2.10.0>`.
+    This is a link to the release notes on the Tarantool documentation website.
+
+The result looks like this:
+
+    Since Tarantool :tarantool-release:`2.10.0`.
+    This is a link to the release notes on GitHub.
+
+    Since Tarantool :doc:`2.10.0 </release/2.10.0>`.
+    This is a link to the release notes on the Tarantool documentation website.
+
+See also the :doc:`guide on writing release notes </contributing/release_notes/>`.
+
+..  _contributing-api-docs_general-description:
+
+Language of the general description
+-----------------------------------
+
+Use one of the two options:
+
+*   Start with a verb in the imperative mood. Example: *Create a fiber.*
+*   Start with a noun. Example: *The directory where memtx stores snapshot files.*
+
+Checklist
+---------
+
+Function or method
+^^^^^^^^^^^^^^^^^^
+
+*   :ref:`Since which Tarantool version <contributing-docs-api_version>`
+*   :ref:`General description <contributing-api-docs_general-description>`
+*   :ref:`Parameters <documenting_parameters>`
+*   What this function returns (if nothing, write 'none')
+*   Return type (if exists)
+*   Possible errors (if exist)
+*   Complexity factors (if exist)
```

Comment:
Can make a separate section below or mention directly here that complexity factors are relevant for a few functions only, and put a link here to https://www.tarantool.io/en/doc/latest/book/box/data_model/#complexity-factors